ratis-server-api/src/main/java/org/apache/ratis/statemachine/StateMachine.java

```diff
@@ -204,10 +205,11 @@ interface LeaderEventApi {
      * This notification is based on "raft.server.rpc.slowness.timeout".
      *
      * @param roleInfoProto information about the current node role and rpc delay information
+     * @param followerInfo information about slow follower
      *
      * @see org.apache.ratis.server.RaftServerConfigKeys.Rpc#SLOWNESS_TIMEOUT_KEY
      */
-    default void notifyFollowerSlowness(RoleInfoProto roleInfoProto) {}
+    default void notifyFollowerSlowness(RoleInfoProto roleInfoProto, FollowerInfo followerInfo) {}
```

Comment:
@inomp3042 , thanks for working on this.  Since FollowerInfo is in the Leader package, it should not appear in the StateMachine API.  How about marking the follower as slow in RoleInfoProto?

Reply:
However, FollowerInfo is in ratis-server-api project's leader package, and `notifyFollowerSlowness`, which is in `LeaderEventApi`, is called on a leader. Even though it is in this case, it should not appear in the StateMachine API?

Reply:
> ... How about marking the follower as slow in RoleInfoProto?

Sorry, this idea is incorrect.  The RoleInfoProto is for leader.  We should not change it.

How about adding a RaftPeerId for the slow follower?
```
    default void notifyFollowerSlowness(RoleInfoProto roleInfoProto, RaftPeerId slowFollower) {}
```

Reply:
> Sorry, this idea is incorrect. The RoleInfoProto is for leader. We should not change it.
How about adding a RaftPeerId for the slow follower?

I see. I think that RaftPeer is better than RaftPeerId. WDYT?

Reply:
Sure, RaftPeer is good.